content/misc/ollama-openwebui/index.md

```diff
@@ -0,0 +1,222 @@
+---
+date: '2024-09-11'
+title: Deploying and Running Ollama and Open WebUI in a ROSA Cluster with GPUs
+tags: ["AWS", "ROSA", "GPU", "Ollama", "OpenWebUI"]
+aliases: ["/docs/misc/ollama-openwebui"]
+authors:
+  - Florian Jacquin
+---
+
+Red Hat OpenShift Service on AWS (ROSA) provides a managed OpenShift environment that can leverage AWS GPU instances. This guide will walk you through deploying Ollama and OpenWebUI on ROSA using instances with GPU for inferences.
+
+## Prerequisites
+
+* A Red Hat OpenShift on AWS (ROSA classic or HCP) 4.14+ cluster
+* OC CLI (Admin access to cluster)
+* ROSA CLI
+
+## Set up GPU-enabled Machine Pool
+
+First we need to check availability of our instance type used here (g4dn.xlarge), it should be in same region of the cluster. Note you can use also Graviton based instance (ARM64) like g5g* but only on HCP 4.16+ cluster.
+
+```bash
+for region in $(aws ec2 describe-regions --query 'Regions[].RegionName' --output text); do
+    echo "Region: $region"
+    aws ec2 describe-instance-type-offerings --location-type availability-zone \
+    --filters Name=instance-type,Values=g4dn.xlarge --region $region \
+    --query 'InstanceTypeOfferings[].Location' --output table
+    echo ""
+done
+```
+
+And then we can create a machine pool with GPU-enabled instances, in our example i use eu-central-1c AZ:
```

Comment:
Suggested change:

"With the region and zone known, now create a machine pool with GPU Enabled Instances. In this example I have used region eu-central-1c"